modules/age.nix

```
@@ -103,9 +116,30 @@ in
       message = "age.sshKeyPaths must be set.";
     }];
 
+    # Create a new directory full of secrets for symlinking (this helps
+    # ensure removed secrets are actually removed, or at least become
+    # invalid symlinks).
+    system.activationScripts.agenixMountSecrets = ''
+      _count="$(basename "$(readlink /run/secrets)" || echo 0)"
+      (( ++_count ))
+      echo "[agenix] symlinking new secrets generation $_count to /run/secrets..."
+      mkdir -p "${cfg.secretsMountPoint}"
+      chmod 0750 "${cfg.secretsMountPoint}"
+      mount | grep "${cfg.secretsMountPoint} type ramfs" -q || mount -t ramfs none "${cfg.secretsMountPoint}" -o nodev,nosuid,mode=0750
+      mkdir -p "${cfg.secretsMountPoint}/$_count"
+      chmod 0750 "${cfg.secretsMountPoint}/$_count"
+      chown :keys "${cfg.secretsMountPoint}" "${cfg.secretsMountPoint}/$_count"
```

Not sure if there are downsides to this, but it seems like it makes sense, since we were running `chown` on a dir and its subdir.

```suggestion
      chown -R :keys "${cfg.secretsMountPoint}"
```

---

The downsides are that it would clobber whatever groups are set in the configuration for e.g. `/run/secrets.d/12/file-with-group-of-foo`, which I would say is 100% undesirable, even if it doesn't affect the current generation (maybe there's some crazy software that caches the real path of a secret file for rereading; it would no longer be able to read it if it relied on `g+r` permissions).

---

I forgot that we removed old secrets generations lol. I guess this wouldn't make a difference at all in that case, but I'd still rather be explicit about which directories / files / etc will be owned by `keys`, and which won't.